experimental/WreathMacdonaldpols/src/WreathMacdonaldpols.jl

```diff
@@ -0,0 +1,241 @@
+
+# Computes the fake degree of a multipartition cf. Ste89 Thm 5.3 and Prop. 3.3.2 Haiman cdm
+function fake_deg(lbb::Multipartition, Q::AbstractAlgebra.Generic.FracField{AbstractAlgebra.Generic.MPoly{QQAbFieldElem{AbsSimpleNumFieldElem}}}, var::AbstractAlgebra.Generic.MPoly{QQAbFieldElem{AbsSimpleNumFieldElem}})
+  r=length(lbb)
+  n=sum(lbb)
+  res=Q(1)
+  for p in 1:n
+    res=res*(1-var^(r*p))
+  end
+  for lambda in lbb
+    for i in 1:length(lambda)
+      for j in 1:lambda[i]
+        hookfactor=(Q(1)-var^(r*(1+lambda[i]+count(>=(j),lambda)-j-i)))
+        res=res//hookfactor
+      end
+    end
+  end
+  res=res*var^b_inv(lbb)
+  return res
+end
+
+# computes C_Delta defined in PhD relation (1.45)
+function C_Delta(r::Int, n::Int, Q::AbstractAlgebra.Generic.FracField{AbstractAlgebra.Generic.MPoly{QQAbFieldElem{AbsSimpleNumFieldElem}}}, var::AbstractAlgebra.Generic.MPoly{QQAbFieldElem{AbsSimpleNumFieldElem}}, mps::Vector{Multipartition{Int}})
+  charTable=character_table_complex_reflection_group(r,1,n)
+  charTirr=[charTable[i,j] for i in 1:nrows(charTable), j in 1:ncols(charTable)]
+  Modules=[multipartition([lbb...]) for lbb in class_parameters(charTable)]
+  l=length(Modules)
+  charTirr=reorder(charTirr,Modules,mps)
+  charTirrT=solve_init(matrix(Q,transpose(charTirr)))
+  rows=zero_matrix(Q,l,0)
+  for i in 1:l
+    col=zero_matrix(Q,l,1)
+    for j in 1:l
+      v=matrix(Q,l,1,map(k->charTirr[i,k]*charTirr[j,k],1:l))
+      x=solve(charTirrT,v,side=:right)
+      col=col+fake_deg(mps[j],Q,var)*x
+    end
+    rows=hcat(rows,col)
+  end
+  return rows
+end
+
+# computes the character of the simple representations of the Cherednik algebra.
+function C_L(r::Int, n::Int, Q::AbstractAlgebra.Generic.FracField{AbstractAlgebra.Generic.MPoly{QQAbFieldElem{AbsSimpleNumFieldElem}}}, var::AbstractAlgebra.Generic.MPoly{QQAbFieldElem{AbsSimpleNumFieldElem}}, mps::Vector{Multipartition{Int}})
+  C_D=C_Delta(r,n,Q,var,mps)
+  d = [var^b_inv(mp)//fake_deg(mp,Q,var) for mp in mps]
+  diag=diagonal_matrix(Q,d)
+  return diag * C_D
+end
+
+function bigger_ord(lbb::Multipartition, wperm::PermGroupElem, coroot::Vector{Int}, mps::Vector{Multipartition{Int}})
+  r=length(lbb)
+  n=sum(lbb)
+  lbbquot=tau_om(lbb,wperm,coroot)
+  res=map(x->x,Iterators.filter(x-> dominates(tau_om(x,wperm,coroot),lbbquot),mps))
+  return res
+end
+
+function smaller_ord(lbb::Multipartition, wperm::PermGroupElem, coroot::Vector{Int}, mps::Vector{Multipartition{Int}})
+  r=length(lbb)
+  n=sum(lbb)
+  lbbquot=tau_om(lbb,wperm,coroot)
+  res=map(x->x,Iterators.filter(x-> dominates(lbbquot,tau_om(x,wperm,coroot)),mps))
+  return res
+end
+
+@doc raw"""
+    wreath_macdonald_polynomials(n::Int,
+                                 r::Int,
+                                 wperm::PermGroupElem,
+                                 coroot::Vector{Int};
+                                 parent::AbstractAlgebra.Generic.MPolyRing{QQAbFieldElem{AbsSimpleNumFieldElem}})
```

Comment:
The kwarg type is overly specific, this should be enough
```suggestion
                                 parent::MPolyRing{<:QQAbFieldElem})
```